File: 2022/d1/d1b.py
import sys

answer=0
total=0
elves=[]

for line in sys.stdin:
    line=line.rstrip()
    if line == "":
       elves.append(total)
       total=0
    else:
        total+=int(line)

elves.sort(reverse=True)

answer=elves[0]+elves[1]+elves[2]

print("The calories of top 3 elves is {ans}".format(ans=answer))

File: 2022/d1/d1b.pl
#!/usr/bin/perl

my $answer=0;
my $total=0;
my @elves=();

while( <STDIN> ) {
    chomp;

    if($_ eq "") {
        push (@elves, $total);
        $total=0;
    } else {
        $total+=$_;
    }
}

@selves=sort { $b <=> $a } @elves;

$answer=$selves[0]+$selves[1]+$selves[2];

printf("The highest calories by any elf is %d\n", $answer);